services/user-svc/internal/organization/organization.go

```diff
@@ -367,57 +367,63 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
 	}, nil
 }
 
-func (s ServiceServer) AcceptInvitation(ctx context.Context, req *pb.AcceptInvitationRequest) (*pb.AcceptInvitationResponse, error) {
+func (s ServiceServer) GetInvitationsByOrganization(ctx context.Context, req *pb.GetInvitationsByOrganizationRequest) (*pb.GetInvitationsByOrganizationResponse, error) {
 	db := hwgorm.GetDB(ctx)
 	log := zlog.Ctx(ctx)
 
-	invitationId, err := uuid.Parse(req.InvitationId)
+	claims, err := common.GetAuthClaims(ctx)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	claims, err := common.GetAuthClaims(ctx)
+	organizationID, err := uuid.Parse(req.OrganizationId)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Check if invite exists
-	currentInvitation, err := GetInvitationByIdAndEmail(db, claims.Email, invitationId)
+	userID, err := common.GetUserID(ctx)
 	if err != nil {
-		if hwgorm.IsOurFault(err) {
-			return nil, status.Error(codes.Internal, err.Error())
-		} else {
-			return nil, status.Error(codes.InvalidArgument, err.Error())
-		}
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if currentInvitation.State == pb.InvitationState_INVITATION_STATE_ACCEPTED {
-		return &pb.AcceptInvitationResponse{}, nil
-	} else if currentInvitation.State != pb.InvitationState_INVITATION_STATE_PENDING {
-		return nil, status.Error(codes.InvalidArgument, "only pending invitations can be accepted")
+	hasAccess, err := IsInOrganization(db, organizationID, userID)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Update invitation state
-	invitation := Invitation{
-		ID: invitationId,
+	if !hasAccess || hwutil.Contains(claims.Organizations, organizationID) {
+		return nil, status.Error(codes.Unauthenticated, "Not a member of this Organization")
 	}
 
-	if err := db.Model(&invitation).Update("state", pb.InvitationState_INVITATION_STATE_ACCEPTED).Error; err != nil {
-		log.Warn().Err(err).Msg("database error")
-		return nil, status.Error(codes.Internal, err.Error())
-	}
+	var invitations []Invitation
+	var invitationsResponse []*pb.GetInvitationsByOrganizationResponse_Invitation
 
-	userID, err := common.GetUserID(ctx)
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+	filter := db.Where("organization_id = ?", organizationID)
+	if req.State != nil {
+		filter = filter.Where("state = ?", req.State)
 	}
 
-	// Add user to organization
-	if err := AddUserToOrganization(ctx, db, userID, currentInvitation.OrganizationID); err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+	if err := filter.Find(&invitations).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			log.Warn().Err(err).Msg("database error")
```

Comment:
```suggestion
```
Not necessary. Every returned error will be logged by our [middleware](https://github.com/helpwave/services/blob/main/libs/common/grpc.go#L285).

Reply:
I just copy pasted it from some other method, we should open an issue to remove those warnings

Reply:
opened #303